rpcserver.go

```diff
@@ -637,6 +637,73 @@ func createVinList(mtx *wire.MsgTx) []btcjson.Vin {
 	return vinList
 }
 
+// createVinList returns a slice of JSON objects for the inputs of the passed
+// transaction.
+func createVinListPrevOut(s *rpcServer, mtx *wire.MsgTx, chainParams *chaincfg.Params, vinExtra int) []btcjson.VinPrevOut {
+	vinList := make([]btcjson.VinPrevOut, len(mtx.TxIn))
+
+	// we don't call fetchInputTransactions() here because it may not be
+	// necessary so we defer until it is.  ( lazy load )
+	var txStore *blockchain.TxStore
```

Comment:
btw, I was doing the fetchInputTransactions() in the loop as an optimization to avoid making that call at all if all inputs are coinbase and don't have prevOuts.   Admittedly this is not the common case and probably rare-ish, but can happen for miners; especially p2pool miners typically have lots of coinbase tx.

I suppose though that in that case, the DB lookup will be pretty fast because it finds no prev transactions.  So I went ahead and moved it to the top as per the @davecgh's suggestion.